Hakawai/Mentions/HKWMentionsPlugin.m

```diff
@@ -254,16 +254,26 @@ - (instancetype)init {
 // Return an array of mentions objects corresponding to the mentions currently in the text view.
 - (NSArray *)mentions {
     NSMutableArray *buffer = [NSMutableArray array];
+    __block HKWMentionsAttribute *previousMention;
 
     [self.parentTextView.attributedText enumerateAttributesInRange:HKW_FULL_RANGE(self.parentTextView.attributedText)
                                                            options:0 usingBlock:^(NSDictionary *attrs, NSRange range, BOOL *stop) {
                                                                id mentionObject = attrs[HKWMentionAttributeName];
                                                                if (![mentionObject isKindOfClass:[HKWMentionsAttribute class]]) {
                                                                    return;
                                                                }
+
                                                                HKWMentionsAttribute *attr = [mentionObject copy];
-                                                               attr.range = range;
-                                                               [buffer addObject:attr];
+                                                               // If two attribute dicts have same mention object and the ranges are touching
+                                                               // they are the same mention and should be combined
+                                                               BOOL isGap = ((previousMention.range.location + previousMention.range.length) < range.location);
```

Comment:
Please add unit tests for this case (gap check logic, as well as the mentions breakdown itself).